sound/soc/sof/ipc4-topology.c

```diff
@@ -128,6 +128,12 @@ static const struct sof_topology_token src_tokens[] = {
 		offsetof(struct sof_ipc4_src, sink_rate)},
 };
 
+/* Core tokens */
+static const struct sof_topology_token core_tokens[] = {
+	{SOF_TKN_COMP_CORE_ID, SND_SOC_TPLG_TUPLE_TYPE_WORD, get_token_u32,
+		offsetof(struct snd_sof_widget, core)},
```

Comment:
can we club this with comp_ext_tokens? I mean the name comp_ext_tokens had a different meaning in IPC3. For IPC4, it could be anything no?

Reply:
yes, good idea.  updated

Reply:
I don't think this is going to work, because  https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/sof/topology.c#L1251-L1264, we use continue for SOF_COMP_EXT_TOKENS, so tokens under this set will not be copied to swidget->tuples.

Reply:
why wont it work @aiChaoSONG. Yes, we use continue but we have already parsed the core and uuis intp swidget on line https://github.com/thesofproject/linux/blob/cfd8c7d5c3e38c0684263bcf9f2580486e5662f6/sound/soc/sof/topology.c#L1253

Reply:
I validated it

Reply:
> why wont it work @aiChaoSONG. Yes, we use continue but we have already parsed the core and uuid into swidget on line
> 
> https://github.com/thesofproject/linux/blob/cfd8c7d5c3e38c0684263bcf9f2580486e5662f6/sound/soc/sof/topology.c#L1253

@ranj063  Yes, it will work, but it also makes the else block in the above code I pasted redundant. maybe we should have a cleanup.

Reply:
yes that wont "work" for IPC4 but it doesnt need to. All we care about is whether swidget->core is what is set in topology or not.